From 1d317faf17d44445309e2f0b215ca5b447e2458b Mon Sep 17 00:00:00 2001
From: Venkateshwar Rao Gannavarapu <venkateshwar.rao.gannavarapu@xilinx.com>
Date: Tue, 27 Oct 2020 15:00:21 -0700
Subject: [PATCH 1734/1852] drm: xlnx: sdi: Add supported EOTF's list to
 connector.

commit dfc8a6c70bfb5821254000b18ac04ed219ffc7e0 from
https://github.com/Xilinx/linux-xlnx.git

This patch adds the supported EOTF's list to connector.
And also fixes the boundary condition of eotf value.

Signed-off-by: Venkateshwar Rao Gannavarapu <venkateshwar.rao.gannavarapu@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_sdi.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_sdi.c b/drivers/gpu/drm/xlnx/xlnx_sdi.c
index 4a44582af49b..54bdf77eae6c 100644
--- a/drivers/gpu/drm/xlnx/xlnx_sdi.c
+++ b/drivers/gpu/drm/xlnx/xlnx_sdi.c
@@ -320,7 +320,7 @@ static void xlnx_sdi_set_eotf(struct xlnx_sdi *sdi)
 
 	eotf = (__u8)frame.eotf;
 
-	if (sdi->prev_eotf == eotf || eotf >= XST352_BYTE2_EOTF_UNKNOWN)
+	if (sdi->prev_eotf == eotf || eotf > XST352_BYTE2_EOTF_UNKNOWN)
 		return;
 
 	switch (eotf) {
@@ -857,6 +857,12 @@ static int xlnx_sdi_create_connector(struct drm_encoder *encoder)
 	xlnx_sdi_drm_connector_create_property(connector);
 	xlnx_sdi_drm_connector_attach_property(connector);
 
+	/* Fill out the supported EOTFs */
+	connector->hdr_sink_metadata.hdmi_type1.eotf |=
+		BIT(V4L2_EOTF_BT_2100_HLG) |
+		BIT(V4L2_EOTF_TRADITIONAL_GAMMA_SDR) |
+		BIT(V4L2_EOTF_SMPTE_ST2084);
+
 	return 0;
 }
 
-- 
2.31.1

